### ISSUE TITLE
Keep multiple cursors for join line command

### ISSUE THREAD

**Issue by failable** [zed-industries/zed#9157]
### Check for existing issues

- [X] Completed

### Describe the feature

Keep multiple cursor for join line command.

### If applicable, add mockups / screenshots to help present your vision of the feature

Zed:
<img width="59" alt="Screenshot 2024-03-11 at 15 19 15" src="https://github.com/zed-industries/zed/assets/5475262/18bba53b-3cbc-4080-8002-5a9938725d63">
<img width="76" alt="Screenshot 2024-03-11 at 15 19 19" src="https://github.com/zed-industries/zed/assets/5475262/bf251863-596d-457d-b0ab-258c2a75ddc5">

Emacs:
<img width="92" alt="Screenshot 2024-03-11 at 15 19 00" src="https://github.com/zed-industries/zed/assets/5475262/683473dd-6fc6-4574-8f17-e7d8be438999">
<img width="78" alt="Screenshot 2024-03-11 at 15 19 04" src="https://github.com/zed-industries/zed/assets/5475262/447d25d7-ec69-43f1-ac6b-05d7c5fb1286">

**Comment by rope-hmg** [comment]
Yes, I would also like this. Sublime Text also keeps the cursors when joining the lines. It's quite useful in a lot of cases.

**Comment by probably-neb** [comment]
Hey! Thanks for reporting this to us. I definitely see how this would be confusing and seems worth looking into more.
We've decided to move feature/enhancement requests to discussions so I'm going to convert this to a discussion, but feel free to continue voicing your desire for this change.
